lib/jobs/ansible-job.js

```diff
@@ -78,10 +95,23 @@ function ansibleJobFactory(
      * @memberOf AnsibleJob
      */
      AnsibleJob.prototype._cleanup = function cleanup() {
+        var self = this;
+        try {
+            var data = fs.readFileSync(self.outFileName, 'utf8');
```

Comment:
It's because you aren't blocking on resolution of the async request I believe, so by the time you update the context, we've already killed the rest of the job and persisted the old context.